rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -311,14 +311,11 @@ namespace rsx
 		draw_state.color_buffer = std::move(copy_render_targets_to_memory());
 		draw_state.depth_stencil = std::move(copy_depth_stencil_buffer_to_memory());
 
-		if (draw_command == rsx::draw_command::indexed)
+		if (draw_state.state.current_draw_clause.command == rsx::draw_command::indexed)
 		{
 			draw_state.vertex_count = 0;
-			for (const auto &range : first_count_commands)
-			{
-				draw_state.vertex_count += range.second;
-			}
-			auto index_raw_data_ptr = get_raw_index_array(first_count_commands);
+			draw_state.vertex_count = draw_state.state.current_draw_clause.get_elements_count();
```

Comment:
Should be draw_state.vertex_count += draw_state.state.current_draw_clause.get_elements_count(); ?